tracing-subscriber/src/registry/sharded.rs

```diff
@@ -89,6 +89,24 @@ fn id_to_idx(id: &Id) -> usize {
     id.into_u64() as usize - 1
 }
 
+pub(crate) struct LayerGuard<'a> {
```

Comment:
I don't think this name really makes it clear what this thing is for. What's a "LayerGuard"? Why "Layer"? I might call it something like `CloseGuard`, and maybe add a comment?

Reply:
I've opted for calling it `CloseGuard` & added a comment describing what this is intended for.